sendrecv/gst/webrtc-sendrecv.c

```diff
@@ -66,15 +66,18 @@ cleanup_and_quit_loop (const gchar * msg, enum AppState state)
 
   if (ws_conn) {
     if (soup_websocket_connection_get_state (ws_conn) ==
-        SOUP_WEBSOCKET_STATE_OPEN)
+        SOUP_WEBSOCKET_STATE_OPEN) {
       /* This will call us again */
       soup_websocket_connection_close (ws_conn, 1000, "");
-    else
+    } else {
       g_object_unref (ws_conn);
+      ws_conn = NULL;
+    }
   }
 
   if (loop) {
     g_main_loop_quit (loop);
+    g_main_loop_unref (loop);
     loop = NULL;
```

Comment:
The proper solution here would probably be to simply quit the main loop and do cleanup/unreffing of it in a single place right after `g_main_loop_run()`. That would keep it all the responsible of a single place in the code instead of having to check in multiple places if it already happened or not.